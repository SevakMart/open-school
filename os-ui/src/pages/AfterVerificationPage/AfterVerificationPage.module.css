.mainContainer {
  text-align: center;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.mainContainer h1 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 800;
  font-size: 2.25rem;
  line-height: 2.75rem;
  color: #222427;
}
.mainContainer p {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 2.5rem;
  color: #5E617B;
}

.closeIcon {
  position: absolute;
  top: 24px;
  right: 24px;
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}
.closeIcon:hover {
  background-color: #4BCEB5;
}/*# sourceMappingURL=AfterVerificationPage.module.css.map */