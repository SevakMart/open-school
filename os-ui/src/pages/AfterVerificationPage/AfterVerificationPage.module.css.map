{"version":3,"sources":["AfterVerificationPage.module.scss","AfterVerificationPage.module.css","../../styles/Mixins/FontMixin.scss","../../styles/Configs/Fonts.scss","../../styles/Configs/Colors.scss","../../styles/Mixins/FlexMixins.scss"],"names":[],"mappings":"AAMA;EACG,kBAAA;EACA,kBAAA;EACA,QAAA;EACA,SAAA;EACA,gCAAA;ACLH;ADMI;EEXF,yBCDY;EDEZ,kBFWwC;EEVxC,gBFUgD;EEThD,kBFSqD;EERrD,oBFQ8D;EACxD,cIXY;AHWpB;ADEI;EEfF,yBCDY;EDEZ,kBFewC;EEdxC,gBFcgD;EEbhD,kBFaqD;EEZrD,mBFY8D;EACxD,cIhBU;AHoBlB;;ADDA;EACI,kBAAA;EACA,SAAA;EACA,WAAA;EACA,sBAAA;EACA,WAAA;EACA,YAAA;EK1BF,aAAA;EACA,mBL0BqB;EKzBrB,mBLyByB;EKxBzB,uBLwBgC;EAC9B,eAAA;ACOJ;ADNI;EACI,yBI9BU;AHsClB","file":"AfterVerificationPage.module.css"}