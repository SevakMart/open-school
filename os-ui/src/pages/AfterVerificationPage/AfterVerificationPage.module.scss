@import "../../styles/Configs/Colors.scss";
@import "../../styles/Configs/Fonts.scss";
@import "../../styles/Mixins/FontMixin.scss";


.mainContainer{
   text-align:center;
   position:absolute;
   top:50%;
   left:50%;
   transform:translate(-50%,-50%);
    h1{
        @include fontStyle($primary-font, normal, 800, 2.25rem, 2.75rem);
        color:$header-main-color;
    }
    p{
        @include fontStyle($primary-font, normal, 600, 1.25rem, 2.5rem);
        color:$main-text-color;
    }
}