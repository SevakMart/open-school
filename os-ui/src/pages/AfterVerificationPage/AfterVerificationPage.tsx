import { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { VERIFICATION_TOKEN_NOT_VALID_MESSAGE_CONTENT, VERIFICATION_TOKEN_NOT_VALID_MESSAGE_HEADER } from '../../constants/Strings';
import { addLoggedInUser } from '../../redux/Slices/loginUserSlice';
import authService from '../../services/authService';
import styles from './AfterVerificationPage.module.scss';

const AfterVerificationPage = () => {
  function getCookie(name: string) {
    const nameEQ = `${name}=`;
    const ca = document.cookie.split(';');
    for (let c of ca) {
      while (c.charAt(0) === ' ') c = c.substring(1, c.length);
      if (c.indexOf(nameEQ) === 0) return c.substring(nameEQ.length, c.length);
    }
    return null;
  }
  const { mainContainer } = styles;
  const [message, setMessage] = useState({ header: 'Please Wait...', content: '' });
  const token = getCookie('verification-token');
  const dispatch = useDispatch();
  const navigate = useNavigate();
  useEffect(() => {
    authService.redirectUserToAccount({ token }).then((response) => {
      if (response.status === 200) {
        dispatch(addLoggedInUser(response));
        navigate('/categories/subcategories');
      } else {
        setMessage({
          header: VERIFICATION_TOKEN_NOT_VALID_MESSAGE_HEADER,
          content: VERIFICATION_TOKEN_NOT_VALID_MESSAGE_CONTENT,
        });
      }
    });
  }, [authService]);

  return (
    <div className={mainContainer}>
      <h1>{ message.header }</h1>
      <p>{ message.content }</p>
    </div>
  );
};
export default AfterVerificationPage;
