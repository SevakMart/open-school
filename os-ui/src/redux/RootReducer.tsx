import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
