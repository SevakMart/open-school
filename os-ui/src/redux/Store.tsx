import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
