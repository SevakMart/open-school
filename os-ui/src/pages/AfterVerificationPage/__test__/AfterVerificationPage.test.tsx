import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../redux/Store';
import AfterVerificationPage from '../AfterVerificationPage';

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
  useLocation: () => ({
    search: '',
  }),
}));
jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create test cases for AfterVerification page', () => {
  test('Make a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <AfterVerificationPage />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
