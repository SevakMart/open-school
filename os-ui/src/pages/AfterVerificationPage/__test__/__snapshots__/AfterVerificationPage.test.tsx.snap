// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for AfterVerification page Make a snapshot test 1`] = `<DocumentFragment />`;
